src/Pos/Ssc/Type.hs

```diff
@@ -0,0 +1,14 @@
+-- | Global export of the @SSC@ type. Exported in order to be able to switch to 
+-- different type-level implementations.
+
+module Pos.Ssc.Type
+       ( SscType
+       ) where
+
+import Pos.Ssc.GodTossing (SscGodTossing)
+--import Pos.Ssc.NistBeacon (SscNistBeacon)
+
+-- | We can also use CPP flags that I proposed that would enable switching between 
+-- implementations without changing the code.
+type SscType = SscGodTossing
```

Comment:
Ok, I'll rename it, but why would you want to have `ssc` without the knowledge of a particular implementation? Seems to me that using this global type seems to fix the type inference and remove the type variable all over the code.